ssh-server/main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/binary"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+
+	"github.com/kr/pty"
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	port = flag.Uint("port", uint(22), "the port to listen on")
+)
+
+func main() {
+	flag.Parse()
+
+	config, listener, err := startServer(*port)
+	if err != nil {
+		log.Fatalf("Failed to start server: %v", err)
+	}
+	log.Printf("Listening on %s...", listener.Addr())
+
+	for {
+		connection, err := listener.Accept()
+		if err != nil {
+			log.Fatalf("Failed to accept incoming connection: %v", err)
+		}
+
+		go func() {
+			handleConnection(connection, config)
+		}()
+	}
+}
+
+func startServer(port uint) (*ssh.ServerConfig, net.Listener, error) {
+	config, err := getConfiguration()
+	if err != nil {
+		return nil, nil, fmt.Errorf("getting configuration: %v", err)
+	}
+
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+	if err != nil {
+		return nil, nil, fmt.Errorf("listen: %v", err)
+	}
+	return config, listener, nil
+}
+
+func getConfiguration() (*ssh.ServerConfig, error) {
+	config := &ssh.ServerConfig{NoClientAuth: true}
```

Comment:
Move this closer to use (right before the call to AddHostKey)

Reply:
done